addons/hr/models/hr_employee.py

```diff
@@ -360,10 +360,32 @@ def _onchange_timezone(self):
             self.tz = self.resource_calendar_id.tz
 
     def _sync_user(self, user, employee_has_image=False):
-        vals = dict(
-            work_contact_id=user.partner_id.id,
-            user_id=user.id,
-        )
+        if not user:
+            existing_partner = self.env['res.partner'].search([
+                ('user_id', '=', False),
+                ('email', '=', self.work_email),
+                ('name', '=', self.name),
+                ], order='id asc', limit=1)
+            if existing_partner:
+                vals = dict(
+                    work_contact_id=existing_partner.id
```

Comment:
if we find something else than his actual work_contact_id, does that mean we'll replace it? Makes no sense IMO

Reply:
The goal here was that after removing the `user_id` (and consequently the `work_contact_id`), we create a new partner that will be tied to the employee record. This new partner record would be identical to the `work_contact_id` record however it would not have a `user_id` relation. The issue I run into here is duplicate res.partner records, one tied to an employee record and another tied to a user (which is kind of not what we want). I'm open to suggestions on how to handle this case better.